packages/core/src/utils/url.ts

```diff
@@ -10,59 +10,69 @@ import type { Dialect, DialectOptions, Options } from '../sequelize';
  * @param connectionString string value in format schema://username:password@host:port/database
  */
 export function parseConnectionString(connectionString: string): Options {
-  const urlParts = url.parse(connectionString, true);
   const options: Options = {};
-  if (urlParts.protocol) {
-    let protocol = urlParts.protocol.replace(/:$/, '');
+
+  // The following connectionStrings are not valid URLs, but they are supported by sqlite.
+  if (connectionString.toLowerCase() === 'sqlite://:memory:' || connectionString.toLowerCase() === 'sqlite::memory:') {
+    options.dialect = 'sqlite';
+    options.host = '';
+
+    return options;
+  }
+
+  const urlObject = new URL(connectionString);
+
+  if (urlObject.protocol) {
+    let protocol = urlObject.protocol.replace(/:$/, '');
     if (protocol === 'postgresql') {
       protocol = 'postgres';
     }
 
     options.dialect = protocol as Dialect;
   }
 
-  if (urlParts.hostname != null) {
-    options.host = urlParts.hostname;
+  if (urlObject.hostname != null) {
+    options.host = urlObject.hostname;
   }
 
-  if (urlParts.pathname) {
-    // decode the URI component from urlParts.pathname value
-    options.database = decodeURIComponent(urlParts.pathname.replace(/^\//, ''));
+  if (urlObject.pathname) {
+    // decode the URI component from urlObject.pathname value
+    options.database = decodeURIComponent(urlObject.pathname.replace(/^\//, ''));
   }
 
-  if (urlParts.port) {
-    options.port = urlParts.port;
+  if (urlObject.port) {
+    options.port = urlObject.port;
   }
 
-  if (urlParts.auth) {
-    const authParts = urlParts.auth.split(':');
-    options.username = authParts[0];
-    if (authParts.length > 1) {
-      options.password = authParts.slice(1).join(':');
-    }
+  if (urlObject.username) {
+    options.username = decodeURIComponent(urlObject.username);
+  }
+
+  if (urlObject.password) {
+    options.password = decodeURIComponent(urlObject.password);
   }
 
-  if (options.dialect === 'sqlite' && urlParts.pathname && !urlParts.pathname.startsWith('/:memory')) {
-    const storagePath = path.join(options.host!, urlParts.pathname);
+  if (options.dialect === 'sqlite' && urlObject.pathname) {
+    const storagePath = path.join(options.host!, urlObject.pathname);
     options.storage = path.resolve(options.storage || storagePath);
   }
 
-  if (urlParts.query) {
+  if (urlObject.searchParams) {
     // Allow host query argument to override the url host.
     // Enables specifying domain socket hosts which cannot be specified via the typical
     // host part of a url.
-    if (urlParts.query.host) {
-      options.host = urlParts.query.host as string;
+    if (urlObject.searchParams.has('host')) {
+      options.host = urlObject.searchParams.get('host')!;
```

Comment:
Without this piece of code, the following unit test is failing;

https://github.com/sequelize/sequelize/blob/7fbaf677aa935fff10beb7286895cf6dec6d83d3/packages/core/test/unit/configuration.test.ts#L195-L201

Reply:
It would, so it would be a breaking change, but this code is a workaround to fix the following issue:

> Enables specifying domain socket hosts which cannot be specified via the typical

Unfortunately we don't have any socket host test, so I don't know what failed to be parsed :(

Reply:
[`ed6d59a` (#16927)](https://github.com/sequelize/sequelize/pull/16927/commits/ed6d59acb52dfb78dcf4904e08856c94a51a6c90) should do the code changes. Will open a documentation PR later

Reply:
Thanks!